core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -449,6 +458,13 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
    */
   protected def doKillExecutors(executorIds: Seq[String]): Boolean = false
 
+  /**
+   * Executor ids which will possibly be preempted by cluster manager.
+   * @return A set of executor IDs
+   */
+  protected def preemptedExecutors: Set[String] = synchronized {
+    preemptedExecutorIDs.toSet
```

Comment:
I see you modify it in the test, but I think that's the wrong thing to do. Better to avoid this transformation or extra allocation or whatever overhead, since this method is called pretty often by the scheduling code.